pkg/controller/job/job_controller.go

```diff
@@ -2122,3 +2127,34 @@ func managedByExternalController(jobObj *batch.Job) *string {
 	}
 	return nil
 }
+
+func keyUIDFunc(obj interface{}) (string, error) {
+	key, err := controller.KeyFunc(obj)
+	if err != nil {
+		return "", err
+	}
+
+	jobObj, ok := obj.(*batch.Job)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
```

Comment:
This code looks duplicated from `deleteJob`. Please extract to a helper function like `asJobObj(obj interface{}) (*batch.Job, error)`